extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -372,73 +371,81 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
+			$key = $this->createKeyForMailNotification( $watcher->getId(), $notification );
 
-				if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
-					$author_signature = $notification->data->msg_author_username;
-				} else {
-					$author_signature = $notification->data->msg_author_displayname .
-						' (' . $notification->data->msg_author_username . ')';
-				}
+			if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
+				$authorSignature = $notification->data->msg_author_username;
+			} else {
+				$authorSignature = $notification->data->msg_author_displayname .
+					' (' . $notification->data->msg_author_username . ')';
+			}
 
-				$data = [];
-				wfRunHooks( 'NotificationGetMailNotificationMessage', [
-					&$notification, &$data, $key, $watcherName, $author_signature, $textNoHtml, $text
-				]);
-				if ( empty( $data ) ) {
-					$data = [
-						'$WATCHER' => $watcherName,
-						'$WIKI' => $notification->data->wikiname,
-						'$PARENT_AUTHOR_NAME' => 	(empty($notification->data->parent_displayname)
-													? ''
-													: $notification->data->parent_displayname),
-						'$AUTHOR_NAME' => $notification->data->msg_author_displayname,
-						'$AUTHOR' => $notification->data->msg_author_username,
-						'$AUTHOR_SIGNATURE' => $author_signature,
-						'$MAIL_SUBJECT' => wfMessage('mail-notification-subject', [
-							'$1' => $notification->data->thread_title,
-							'$2' => $notification->data->wikiname
-						])->text(),
-						'$METATITLE' => $notification->data->thread_title,
-						'$MESSAGE_LINK' =>  $notification->data->url,
-						'$MESSAGE_NO_HTML' =>  $textNoHtml,
-						'$MESSAGE_HTML' =>  $text,
-						'$MSG_KEY_SUBJECT' => $key,
-						'$MSG_KEY_BODY' => 'mail-notification-body',
-						'$MSG_KEY_GREETING' => 'mail-notification-html-greeting',
-					];
-				}
+			$watcherName = $watcher->getName();
+			$data = [];
+			wfRunHooks( 'NotificationGetMailNotificationMessage', [
+				&$notification, &$data, $key, $watcherName, $authorSignature, $textNoHtml, $text
+			] );
+
+			if ( !empty( $data ) ) {
+				// The old email delivery has been kept here for forums, which use the hook above.
+				// This logic can be removed when forum emails are migrated to the new email extension
+				// or when forums are retired.
+				$this->sendEmail( $watcher, $data );
+			} else {
+				$params = [
+					'targetUser' => $watcherName,
+					'fromAddress' => $watcher->getEmail(),
+
+					'wallUserName' => $notification->data->wall_username,
+					'threadUserName' => $notification->data->parent_username,
+					'replyUserName' => $notification->data->msg_author_username,
+					'isThread' => $notification->isMain(),
+
+					'messageTitle' => $notification->data->thread_title,
+					'messageBody' => $textNoHtml,
+					'messageUrl' => $notification->data->url
+				];
 
-				if ( !( $watcher->getBoolOption('unsubscribed') === true ) ) {
```

Comment:
Yes, I removed because @jsutterfield noted we already do this in EmailController but we need it here until we move all emails from this class over to the new system.